sky/utils/common_utils.py

```diff
@@ -755,3 +756,40 @@ def is_port_available(port: int, reuse_addr: bool = True) -> bool:
             return True
         except OSError:
             return False
+
+
+# TODO(aylei): should be aware of cgroups
+def get_cpu_count() -> int:
+    """Get the number of CPUs.
+
+    If the API server is deployed as a pod in k8s cluster, we assume the
+    number of CPUs is provided by the downward API.
+    """
+    cpu_count = os.getenv('SKYPILOT_POD_CPU_CORE_LIMIT')
```

Comment:
Can be another PR, we should actually test this with the real helm deployment with request limit. (We need to add back the resource limit arguments in our helm chart in this commit: https://github.com/skypilot-org/skypilot/pull/4737/files/b973f2b99b18ac0b749afec7a7a73cfc8190cfc4#diff-de8be15ab76152d30fe346b511ac51383d8475c1b12615a5a3765a43a7ccd304

Reply:
These functions are moved from [sky/server/requests/executor.py](https://github.com/skypilot-org/skypilot/pull/4731/files#diff-2731eb29fca4c252c0eaaa23e35f911c7448e882ee7e4e65b02469eda220da09) without modification to get the right resource limit when launching the api-server.

Agreed we should test with real helm deployment, will take a look and refine the helm chart in another PR